Replacing the static content of the Challenge page

elfeto opened the pull request:
Here I replace the static content of the static page, and also add the
published papers and contributed software section.

For this to work, there has to be some manual labor done.

For the software section, the folder 'sources' has to exists under the project.
If the folder exists, it will create a URL under the challenge index page to a
Nginx http index page, just like in the old website.

For the papers, is somewhat similar.
- Create a papers folder and put all of the papers into the folder (e.g. like the papers folder for https://physionet.org/content/challenge-2018/)
- Save https://archive.physionet.org/challenge/2018/papers/ as an HTML file and upload the file to the challenge files named 'index.html' and place it in the same folder.

Here I fix #819

bemoody commented:
This sounds like an excellent change (haven't yet tested it.)

I'm not keen on hardcoding things based on regular expression matches
of project names.  On the other hand, I think it's fine to check for
existence/nonexistence of hardcoded file names. :)

We probably want to restrict the PublishedProjects by is_latest_version=True.

I think other challenges besides the CinC ones *should* be listed
here, though perhaps grouped separately.

Please add some demo data, if necessary, so that we can test the
functionality of this page in the development server.

These are all minor issues, though (and as I said, I haven't tested
the code), so @tompollard, if it works and you're happy with it, feel
free to merge.

tompollard commented:
Thanks @bemoody. I have copied the papers and code across to the project folders on production, and created most of the index files. Will finish the files off, do some testing and then hopefully we'll be ready to merge :)

tompollard commented:
The index files for papers are below. I put them together pretty quickly and haven't checked carefully, so I'm sure there are some bugs (e.g. broken links) that need fixing. 

- https://physionet.org/files/challenge-2001/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2001/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2001/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2001/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2001/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2001/1.0.0/papers/index.html

tompollard commented:
The index files for papers are below. I put them together pretty quickly and haven't checked carefully, so I'm sure there are some bugs (e.g. broken links) that need fixing. 

- https://physionet.org/files/challenge-2000/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2001/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2002/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2003/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2004/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2005/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2006/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2007/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2008/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2009/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2010/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2011/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2012/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2013/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2014/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2015/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2016/1.0.0/papers/index.html
- https://physionet.org/files/challenge-2017/1.0.0/papers/index.html

tompollard commented:
Thanks Felipe, this is a nice, clean solution, and I can't think of a way to avoid a certain amount of hardcoding for now. I've list a few minor things below (incorporating @bemoody's points above). Please could you respond to them so that we can merge?

#### Should be fixed before we merge

- Restricting the PublishedProjects by `is_latest_version=True` is a good idea, to avoid the same challenge being listed multiple times.
- The text link text is listed under the wrong column at the moment:

<img width="381" alt="Screen Shot 2019-11-22 at 15 12 55" src="https://user-images.githubusercontent.com/822601/69458810-39f96300-0d3e-11ea-9e33-f16f82e532c6.png">

#### Nice to address now, but could wait (as far as I'm concerned...if someone disagrees, please shout!)

- Add some demo data, if necessary, so that we can test the functionality of this page in the development server.

#### To address later (raised as new issues if necessary):

- Benjamin's point that: "I think other challenges besides the CinC ones *should* be listed here, though perhaps grouped separately". Let's keep them together for now, and perhaps separate later.
- We may want to support an index.html for the contributed software too. There a lot of inconsistencies in the way project pages have been handled over time, and some of the new projects include a summary page (e.g. https://archive.physionet.org/challenge/2017/sources/) .
- Form validation for publishing projects doesn't allow us to create a slug like `challenge-nnnn` (which is the slug used for all legacy challenges):

<img width="748" alt="Screen Shot 2019-11-22 at 15 27 36" src="https://user-images.githubusercontent.com/822601/69459101-ff43fa80-0d3e-11ea-9af9-7d34cdb74e59.png">

elfeto commented:
I added `is_latest_version=True`.
Fixed the order of papers and software, and added the index in case the index.html exists in software. 
The slug is fixed in a different PR I made.

I will try to add challenge demo data without files between today and tomorrow. If you merge, i can do another PR.

tompollard commented:
Thanks, looks great (and the optional `index.html` for software is perfect). Sorry, a couple more quick thoughts before we merge:

- `sources` is a slightly odd folder name for contributed software, but I assume this is consistent with what was used in the past? If so, let's keep it as that.
- Is there any sorting on the order of the challenges? I don't think so, but I might be wrong. If not, please could you sort by date (newest at top)?

elfeto commented:
> sources is a slightly odd folder name for contributed software, but I assume this is consistent with what was used in the past? If so, let's keep it as that.

Yes, and I just did the sorting now.

tompollard commented:
thanks!

tompollard commented:
@matthewreyna the challenge page now lists papers and software https://physionet.org/about/challenge/
